### PR TITLE
Modify readme

### PR DESCRIPTION
**Pull request by zoey1124**
* modified README.md, write more specific instructions
* modified autrite/pipeline.py, fix an imported package bug to make it runnable

### CODE REVIEW

File: autrite/pipeline.py [hunk 1]
```diff
@@ -41,6 +41,7 @@
         print("[Before filtering DB constraints] ", len(constraints))
         constraints = [c for c in constraints if c.db == True]
         print("[After filtering DB constraints] ", len(constraints))
+    exit(0)
```
Comment (LiuXiaoxuanPKU) [inline autrite/pipeline.py:44]: why exit(0) here?

File: README.md [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
 ```
-
 ### Run tests
-under constr_extractor folder, run
-```
-rspec spec/extractor_spec.rb
-```
+For running constrains extraction tests, run `rspec spec/extractor_spec.rb` under constr_extractor folder. All code sections take care of constraints extraction is under `/constropt/constr_extractor/` directory. 
 
 ## Query Rewriter
```
Comment (LiuXiaoxuanPKU) [inline README.md:16]: Have you checked if the statement here is correct? Are those tests runnable?

File: README.md [hunk 3]
```diff
@@ -1,21 +1,21 @@
 # ConstrOpt
+ConstrOpt is a library that leverages application data constraints to optimize database-backed web applications. You can find the related paper [here](https://arxiv.org/abs/2205.02954).
+
+## Run Experiment
```
Comment (LiuXiaoxuanPKU) [inline README.md:4]: It's better to make appname a parameter

**Comment by LiuXiaoxuanPKU** [comment]
Also, some examples of READNE.md
https://github.com/ucbrise/actnn
https://github.com/LiuXiaoxuanPKU/GACT-ICML

File: autrite/pipeline.py (hunk 1 above)
Comment (zoey1124) [reply autrite/pipeline.py]: it's a bug, my bad. Now fixed.

File: README.md (hunk 3 above)
Comment (zoey1124) [reply README.md:5]: I'm not sure what you mean here, do you mean make "ConstrOpt" shown as code format? (e.g. `ConstrOpt`)

Comment (LiuXiaoxuanPKU) [reply README.md:5]: Can you change pipeline instruction to `python3 autrite/pipeline.py appname`?

Comment (zoey1124) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (zoey1124) [reply README.md:63]: yes, checked.

File: autrite/pipeline.py (hunk 1 above)
Comment (LiuXiaoxuanPKU) [reply autrite/pipeline.py]: Why you add from `Provedumper import ProveDumper` here?

File: README.md [hunk 4]
```diff
@@ -1,9 +1,28 @@
...
 Here we present `ConstrOpt`, the first tool that identifies data relationships by analyzing the programs that generate and maintain the persist data. Once identified,`ConstrOpt` leverages the found constraints to optimize the application's physical design and query execution by rewriting queries. Instead of developing a fixed set of predefined rewriting rules, `ConstrOpt` employs an enumerate-test-verify technique to automatically exploit the discovered data constraints to improve query execution. 
-![experiment image](https://github.com/LiuXiaoxuanPKU/ConstrOpt/blob/main/figures/7.2/redmine.pdf)
+## Install
+- Install third party libraries for ruby: under `constr_extractor` folder, run
+``` 
+bundler insatll
+```
+- Install python dependencies: we use [microsoft z3](https://github.com/Z3Prover/z3) solver to prove sql equivalence. use `pip3 install <package name>` to install the following python3 dependencies. E.g., `pip3 install z3-solver`.
```
Comment (LiuXiaoxuanPKU) [inline README.md:11]: it's better to put all the packages in the requirements.txt and pip install -r requirements.

File: README.md [hunk 5]
```diff
@@ -18,18 +37,33 @@ The following is the version information we use for apps in our experiments.
...
+- Numerical: the value of a numerical field must lie in the range
+specified in the application code.
+- Foreign key: same as the SQL foreign key constraint, where the
+field points to the primary key of the referenced table.
+#### Run tests
+Under the root directory (`\ConstrOpt`), following the following command:
+```
+cd constropt/constr_extractor/      # enter directory
+bundle install                      # install ruby dependencies
+chmod +x run_test.sh                # give permisson to this script
```
[the remark this answers is not in the document]
Comment (zoey1124) [reply README.md]: fixed

File: README.md (hunk 4 above)
Comment (zoey1124) [reply README.md]: requirements.txt done and pushed

File: autrite/pipeline.py (hunk 1 above)
Comment (zoey1124) [reply autrite/pipeline.py]: cuz we use `Provedumper` [here](https://github.com/LiuXiaoxuanPKU/ConstrOpt/blob/da1927b608ecb8b92dd9e46d15eaa99a217c825b/constropt/autrite/pipeline.py#L139).

**Pull request merged**